backend/wmg/pipeline/summary_cubes/calculate_markers.py

```diff
@@ -53,50 +58,20 @@ def wrapped(*args, **kwargs):
 
     return wrapped
 
+    # this decorator function is required to allow lru_cache to
+
 
-# this decorator function is required to allow lru_cache to
 # memoize the function as dicts, lists, and WmgSnapshot objects
 # are not typically hashable.
 @_make_hashable
 @lru_cache(maxsize=None)
-def _query_tiledb(
-    filters: dict,
+def _query_tiledb_context_memoized(
```

Comment:
This refactor makes it so that we only need to do one _memoized_ tiledb query (for the context population). The data for the target population is by definition present in the context data, so we can extract the target data from the context query result.